### PR TITLE
[WOE] Implement Curse of the Werefox

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Curse of the Werefox]]

**Comment by Susucre** [comment]
Oh right. I thought about the target be invalidated (the entire effect fizzle), but not that the Role could potentially not be created after that.

**Comment by Susucre** [comment]
@theelk801 I have changes that should be ready for 

However there is a more general bug that Roles Tokens are created on permanents that have protection from them, they are created attached, then instantly fall off:
image
Seems that needs a more general fix. Did not try protection from colorless but that should be the same.

**Comment by Susucre** [comment]
Made a standalone issue for the bug #11022

File: Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        Permanent target = game.getPermanent(source.getFirstTarget());
+        if (target == null) {
+            return false;
+        }
+
+        boolean didCreate =
+                new CreateRoleAttachedTargetEffect(RoleType.MONSTER)
+                        .setTargetPointer(new FixedTarget(target, game))
+                        .apply(game, source);
+        if (!didCreate) {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java:72]: are you sure checking the return value of the apply method actually works how you want it to?

File: Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java [hunk 2]
```diff
@@ -89,6 +91,10 @@ public boolean canTarget(MageObject source, Game game) {
                 return !((FilterCard) filter).match((Card) source, ((Permanent) source).getControllerId(), this, game);
             } else if (source instanceof Card) {
                 return !((FilterCard) filter).match((Card) source, ((Card) source).getOwnerId(), this, game);
+            } else if (source instanceof Token) {
+                // Fake a permanent with the Token info.
+                PermanentToken token = new PermanentToken((Token) source, null, game);
```
Comment (Susucre) [inline Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java:96]: @theelk801 @xenohedron So this is almost good enough. The only part not correct is that I do not have the player that will put the token into play.
So I'm afraid the protection from player cards are escaping some of the filtering due to using the 2 arguments `match`.

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java:96]: I think this looks good to me but I'll leave it to @xenohedron to approve

Comment (Susucre) [reply Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java:96]: I think I'll still need to add something for protection from player down below.
Like If you [[Return Triumphant]] a [[True-Name Nemesis]], and somehow choose yourself for the protection.

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java:96]: I think that can be done separately as it's something that matters for any aura token creation

Comment (Susucre) [reply Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java:96]: Alright, Yenna also has issues on that regard, but due to aura token using their raw target and not caring for protection. Need to think that through too.

File: Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            return false;
+        }
+
+        ReflexiveTriggeredAbility ability = new ReflexiveTriggeredAbility(
+                new FightTargetsEffect(false), false,
+                "that creature fights up to one target creature you don't control"
+        );
+        ability.addTarget(source.getTargets().get(0));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java:80]: You should use `setTargetPointer` here instead as the the trigger only actual targets one permanent

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java (hunk 3 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java:80]: I have no idea how to do that while still using FightTargetsEffect.
That one effect really wants to have two targets set.

Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java:80]: I'm pretty sure the fight effect has code to handle having a fixed target

Comment (Susucre) [reply Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java:80]: well don't think so.
I'll do it like Tolsimir, Friend of Wolves or Gruul Ragebeast, and define a custom `Effect` that calls `Permanent::fight`.

File: Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        if (!didCreate) {
+            return false;
+        }
+
+        ReflexiveTriggeredAbility ability = new ReflexiveTriggeredAbility(
+                new CurseOfTheWerefoxFightEffect(), false,
+                "that creature fights up to one target creature you don't control"
+        );
+        ability.getEffects().setTargetPointer(new FixedTarget(target.getId(), game));
+        ability.addTarget(new TargetOpponentsCreaturePermanent(0, 1));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/CurseOfTheWerefox.java:82]: Opponent controls isn't the same as you don't control

TargetCreaturePermanent(StaticFilters.FILTER_CREATURE_YOU_DONT_CONTROL)

**Review by xenohedron — COMMENTED** [review]
Minor adjustment but otherwise this looks good to me

**Pull request merged**